src/cript/nodes/exceptions.py

```diff
@@ -39,3 +39,25 @@ def __init__(self, node_type: str, json_dict: str):
 
     def __str__(self):
         return f"JSON Serialization failed for node type {self.node_type} with JSON dict: {self.json_str}"
+
+
+class UneditableAttributeError(Exception):
```

Comment:
because then you can catch this exception and handle it.

Reply:
You can catch it, but what can you handle it? it looks like you will just ingore them. There is no information about the attribute involved (name, current_value, desired_value, reason).

Reply:
what are we thinking here?

return an `UneditableAttributeError` when the user tries to modify a node attribute that they do not have permission to modify such as `Group` or `User` or remove the setter and throw a python `AttributeError`?

I think both could work. I lean towards `UneditableAttributeError` because I think that is more user friendly, but we can do either for now and perfect it later

Reply:
See my comment above, in how we could handle this in a generic way.

Reply:
I am a big advocate for accurate, understandable, and timely user feedback.

I updated the code and removed all the setters to for `Group` and `User` nodes for now. This will trigger an `AttributeError` if the user attempts to set it. For now, I think this is good enough, so that we can play around with it and debug it, but in future updates we probably want to come back and update all read-only attributes to be more user friendly.

I wrote an [issue](https://github.com/C-Accel-CRIPT/Python-SDK/issues/45) for it so we do not forget about it.